source/common/filter/tcp_proxy.cc

```diff
@@ -140,6 +140,10 @@ void TcpProxy::readDisableUpstream(bool disable) {
 
 void TcpProxy::readDisableDownstream(bool disable) {
   read_callbacks_->connection().readDisable(disable);
+  if (!config_) {
```

Comment:
My bad. I should have added a comment. I am reusing the tcp proxy code in ws_handler_impl, where I initialize the tcp_proxy class with a null config.